services/src/unit-test/java/uk/gov/moj/sdt/services/utils/MessagingUtilityTest.java

```diff
@@ -0,0 +1,89 @@
+import uk.gov.moj.sdt.domain.api.IBulkSubmission;
+import uk.gov.moj.sdt.domain.api.IIndividualRequest;
+import uk.gov.moj.sdt.domain.api.ITargetApplication;
+import uk.gov.moj.sdt.services.messaging.MessageWriter;
+import uk.gov.moj.sdt.services.messaging.SdtMessage;
+import uk.gov.moj.sdt.services.messaging.api.IMessageWriter;
+import uk.gov.moj.sdt.utils.AbstractSdtUnitTestBase;
+import uk.gov.moj.sdt.utils.transaction.synchronizer.MessageSynchronizer;
+import uk.gov.moj.sdt.utils.transaction.synchronizer.api.IMessageSynchronizer;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class MessagingUtilityTest extends AbstractSdtUnitTestBase {
+
+    @BeforeEach
+    @Override
+    public void setUp() {
+    }
+
+    @Test
+    public void testEnqueueRequestNew() {
+        // Given
+        IIndividualRequest individualRequest = mock(IIndividualRequest.class);
+        IBulkSubmission bulkSubmission = mock(IBulkSubmission.class);
+        ITargetApplication targetApplication = mock(ITargetApplication.class);
+        IMessageSynchronizer messageSynchronizer = mock(IMessageSynchronizer.class);
+        IMessageWriter messageWriter = mock(IMessageWriter.class);
+
+        when(individualRequest.getBulkSubmission()).thenReturn(bulkSubmission);
+        when(bulkSubmission.getTargetApplication()).thenReturn(targetApplication);
+        when(targetApplication.getTargetApplicationCode()).thenReturn("targetAppCode");
+        when(individualRequest.getSdtRequestReference()).thenReturn("sdtRequestReference");
+
+        MessagingUtility messagingUtility = new MessagingUtility(messageWriter,messageSynchronizer);
+        messagingUtility.setMessageSynchronizer(messageSynchronizer);
+        messagingUtility.setMessageWriter(messageWriter);
+
+        ArgumentCaptor<Runnable> runnableCaptor = ArgumentCaptor.forClass(Runnable.class);
+
+        //When
+        messagingUtility.enqueueRequest(individualRequest);
+
+        // Then
+        verify(messageSynchronizer).execute(runnableCaptor.capture());
+        Runnable runnable = runnableCaptor.getValue();
+        runnable.run();
+
+        verify(messageWriter).queueMessage(any(SdtMessage.class), eq("targetAppCode"), eq(false));
+    }
+
+
+
+    @Test
+    void testMessageSynchronizer() {
+        IMessageWriter messagingWriter = mock(MessageWriter.class);
+        IMessageSynchronizer messageSynchronizer = mock(MessageSynchronizer.class);
+        MessagingUtility messagingUtility = new MessagingUtility(messagingWriter,messageSynchronizer);
+
+        messagingUtility.setMessageSynchronizer(messageSynchronizer);
+
+        assertNotNull(messagingUtility.getMessageSynchronizer(),"should have returned MessageSync");
```

Comment:
Better to use an assertEquals() here to prove that the set method has worked.

Reply:
done